hub/saml-engine/src/main/java/uk/gov/ida/hub/samlengine/SamlEngineModule.java

```diff
@@ -656,10 +656,15 @@ private DecoratedSamlResponseToIdaResponseIssuedByIdpTransformer getResponseToIn
                 hubEntityId);
     }
 
+    @Provides
+    @Singleton
+    private SigningCertFromMetadataExtractor getSigningCertFromMetadataExtractor(@Named(VERIFY_METADATA_RESOLVER) MetadataResolver metadataResolver, @Named("HubEntityId") String hubEntityId) throws ComponentInitializationException {
```

Comment:
Just to make you aware of this --> We are currently using the same stuff signing keys to sign both Verify and eIDAS AuthnRequest. 

If this changes in the future (i.e. if we decide that the Connector should sign with a different key), this logic will break.

Reply:
Also to add - it is actually a bit confusing. 

It's explicitly looking at the federation's metadata and key to populate the eIDAS's AuthnRequest as a Connector. 

It works because of the current design but logically it doesn't quite makes sense...